### ISSUE TITLE
Fix substitutions eating characters with selections

### ISSUE THREAD

**Issue by ajzafar** [garbas/vim-snipmate#302]
Title says it all. They kind of work but not in 100% of cases. Need a sample snippet of exactly where they go wrong.

Originated from #265.

**Comment by ajzafar** [comment]
The issue I had found with evaluations was actually related to substitutions, not evaluations. If a selection is made that is longer than the first option when the selection stop is first entered, then substitutions will eat characters. Skipping past the selection then revisiting it does not seem to result in this bug.

```
snippet test
        ${1|foo|foobar|foobarbaz} ${1/[oa]/x} _____ ${2}
```

I've updated the issue title.

**Comment by ajzafar** [comment]
This is fixed in d887603. Turns out substitutions are not related at all.

Evaluations are now supported as well.
